smartsim/_core/control/controller.py

```diff
@@ -275,7 +341,7 @@ def init_launcher(self, launcher: str) -> None:
         else:
             raise TypeError("Must provide a 'launcher' argument")
 
-    def _launch(self, manifest: Manifest) -> None:
+    def _launch(self, manifest: Manifest) -> LaunchedManifest[t.Tuple[str, Step]]:
```

Comment:
out-of-scope enhancement request: I think you're potentially implementing exactly what we might need for an upcoming launch preview ticket. 

Could we save some time later by separating the launch manifest building from the "true launch?"

@amandarichardsonn might be interested in this work